### PR TITLE
Firing pin extraction device

### PR DESCRIPTION
**Pull request by Sindorman**
- Adds firing pin extraction device, sprites provided by Kyres

- RD's locker now has firing pin extraction device.

### IC changelog
`NT weapons research division decided to provide all Research Directors with a tool that can extract firing pins without damaging them. These devices are only approved to be used by Head of Staff, specifically Research Director. Any attempt of reproducing device will be considered as illegal action.`

### CODE REVIEW

**Comment by Arrow768** [comment]
@OneOneThreeEight Didnt you want to do something similar aswell ?

**Comment by OneOneThreeEight** [comment]
I gave them the go-ahead since they already invested an amount of effort into this, I ended up diverging from what I wanted to do as well.

**Comment by ParadoxSpace** [comment]
let research print it with illegal tech

**Comment by Sindorman** [comment]
Yeah, I will add it

**Comment by Sindorman** [comment]
@ParadoxSpace @skull132 Request addressed, tested and ready for merge.

File: code/game/objects/items/devices/pin_extractor.dm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/obj/item/device/pin_extractor/update_icon()
+	if(on)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
+/obj/item/device/pin_extractor/attack_self(mob/user)
+	on = !on
+	if(on && activation_sound)
```
Comment (Arrow768) [inline code/game/objects/items/devices/pin_extractor.dm]: What is the purpose of having to turn it on, before you can use it ?
What prevents you from leaving it on at all times ?

Comment (Sindorman) [reply code/game/objects/items/devices/pin_extractor.dm]: Nothing, just a thing to have animation working.

File: code/game/objects/items/devices/pin_extractor.dm [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+/obj/item/device/pin_extractor/afterattack(atom/target, mob/user, proximity_flag)
+	if(proximity_flag)
+		if(istype(target, /obj/item/weapon/gun))
+			if(!on)
+				to_chat(user, "<span class ='notice'>\The [src.name] is not turned on.</span>")
+				..()
+				return
+			var/obj/item/weapon/gun/G = target
+			if(G.pin)
+				to_chat(user, "<span class ='notice'>You begin removing [G.name]'s [G.pin.name] using \the [src.name], it will take 5 seconds.</span>")
```
[the remark this answers is not in the document]
Comment (ParadoxSpace) [reply code/game/objects/items/devices/pin_extractor.dm]: What screwing is unwanted?

**Pull request merged**